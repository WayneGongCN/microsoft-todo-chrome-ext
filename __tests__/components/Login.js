import React from 'react';
import renderer from 'react-test-renderer';
import { render, queryByAttribute, fireEvent } from '@testing-library/react'
import Login from '../../src/components/Login';


describe('Component: <Login />', () => {
  test('<Login /> Shoud show login button', () => {
    const mockLogin = jest.fn()
    const component = renderer.create(<Login loading={false} login={mockLogin} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('<Login /> Shoud show loading buttong when loading is true', () => {
    const mockLogin = jest.fn()
    const component = renderer.create(<Login loading={true} login={mockLogin} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })

  test('<Login /> Click login button shoud call login props', () => {
    const mockLogin = jest.fn()
    const dom = render(<Login loading={false} login={mockLogin}></Login>)
    const loginBtn = queryByAttribute('id', dom.container, 'login-btn')
    fireEvent.click(loginBtn)
    expect(mockLogin.mock.calls.length).toBe(1)
  })

  test(`<Login /> When loading state can't click login button`, () => {
    const mockLogin = jest.fn()
    const dom = render(<Login loading={true} login={mockLogin}></Login>)
    const loginBtn = queryByAttribute('id', dom.container, 'login-btn')
    fireEvent.click(loginBtn)
    expect(mockLogin.mock.calls.length).toBe(0)
  })
})
