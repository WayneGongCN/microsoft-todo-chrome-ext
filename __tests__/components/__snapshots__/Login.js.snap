// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Login /> <Login /> Shoud show loading buttong when loading is true 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
  disabled={true}
  id="login-btn"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    Please waiting ...
    <span
      className="MuiButton-endIcon MuiButton-iconSizeSmall"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": 20,
            "width": 20,
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={20.2}
            strokeWidth={3.6}
            style={Object {}}
          />
        </svg>
      </div>
    </span>
  </span>
</button>
`;

exports[`Component: <Login /> <Login /> Shoud show login button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
  disabled={false}
  id="login-btn"
  onBlur={[Function]}
  onClick={[MockFunction]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    LOGIN
  </span>
</button>
`;
